src/test/java/uk/ac/ebi/ampt2d/accession/commons/core/BasicAccessioningServiceSaveDelegateTest.java

```diff
@@ -0,0 +1,106 @@
+/*
+ *
+ * Copyright 2018 EMBL - European Bioinformatics Institute
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+package uk.ac.ebi.ampt2d.accession.commons.core;
+
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.boot.test.autoconfigure.orm.jpa.DataJpaTest;
+import org.springframework.test.annotation.Commit;
+import org.springframework.test.context.ContextConfiguration;
+import org.springframework.test.context.junit4.SpringRunner;
+import org.springframework.test.context.transaction.TestTransaction;
+import uk.ac.ebi.ampt2d.accession.commons.generators.SingleAccessionGenerator;
+import uk.ac.ebi.ampt2d.accession.commons.generators.exceptions.AccessionCouldNotBeGeneratedException;
+import uk.ac.ebi.ampt2d.accession.commons.hashing.SHA1HashingFunction;
+import uk.ac.ebi.ampt2d.accession.commons.persistence.BasicSpringDataRepositoryDatabaseService;
+import uk.ac.ebi.ampt2d.test.TestModel;
+import uk.ac.ebi.ampt2d.test.configuration.TestDatabaseServiceTestConfiguration;
+import uk.ac.ebi.ampt2d.test.persistence.TestEntity;
+import uk.ac.ebi.ampt2d.test.persistence.TestRepository;
+
+import java.util.HashMap;
+import java.util.Map;
+
+@RunWith(SpringRunner.class)
+@DataJpaTest
+@ContextConfiguration(classes = {TestDatabaseServiceTestConfiguration.class})
+public class BasicAccessioningServiceSaveDelegateTest {
+
+    @Autowired
+    private TestRepository repository;
+
+    @Autowired
+    private BasicSpringDataRepositoryDatabaseService<TestModel, TestEntity, String, String> databaseService;
+
+    @Test
+    public void testUpdateWhenHashAndAccessionCoincide() throws AccessionCouldNotBeGeneratedException {
```

Comment:
I don't see what is being tested in this code. According to the title, I expect that an update will be made in some moment, but I don't see where is this being asserted.